IdentityCore/src/cache/ios/MSIDKeychainTokenCache.m

```diff
@@ -22,30 +22,218 @@
 // THE SOFTWARE.
 
 #import "MSIDKeychainTokenCache.h"
+#import "MSIDTokenCacheKey.h"
+#import "MSIDTokenSerializer.h"
+#import "MSIDKeychainUtil.h"
+#import "MSIDToken.h"
+#import "MSIDError.h"
+
+static NSString *const s_libraryString = @"MSOpenTech.ADAL.1";
+static NSString *const s_wipeLibraryString = @"Microsoft.ADAL.WipeAll.1";
+static MSIDKeychainTokenCache *s_defaultCache = nil;
+static NSString *s_defaultKeychainGroup = @"com.microsoft.adalcache";
+
+@interface MSIDKeychainTokenCache ()
+
+@property (nonnull) NSString *keychainGroup;
+
+@end
 
 @implementation MSIDKeychainTokenCache
 
-- (BOOL)setItemWithKey:(MSIDTokenCacheKey *)key
-                  item:(MSIDToken *)item
-            serializer:(id<MSIDTokenSerializer>)serializer
-               context:(id<MSIDRequestContext>)context
-                 error:(NSError **)error
+#pragma mark - Public
+
++ (NSString *)defaultKeychainGroup
 {
-    return NO;
+    return s_defaultKeychainGroup;
+}
+
++ (void)setDefaultKeychainGroup:(NSString *)defaultKeychainGroup
+{
+    if (s_defaultCache)
+    {
+        MSID_LOG_ERROR(nil, @"Failed to set default keychain group, default keychain cache has already been instantiated.");
+        
+        @throw @"Attempting to change the keychain group once AuthenticationContexts have been created or the default keychain cache has been retrieved is invalid. The default keychain group should only be set once for the lifetime of an application.";
```

Comment:
Do we actually need to throw here?

Reply:
We throw in ADAL (I copied the logic to MSID). Do you suggest just to ignore and do nothing in this case instead of throwing?

Reply:
I'm fine with it, if we throw in ADAL. Just wanted to verify we don't break anything existing here.